ingests/shift.py

```diff
@@ -0,0 +1,357 @@
+        f"{SHIFT_URL}energy/{energy_source}?chart-type=line&chart-types=line&chart-types=ranking&"
+        f"disable-en=false&energy-unit={ENERGY_UNIT}"
+    )
+    # Add each country to the url.
+    for country in countries:
+        query_url += f"&group-names={country.replace(' ', '%20').replace('&', '%26')}"
+    # Add some conditions to the query (not all of them may be necessary).
+    query_url += (
+        f"&is-range=true&dimension=total&end={END_YEAR}&start={START_YEAR}&multi=true&type=Production&"
+        f"import-types=Imports&import-types=Exports&import-types=Net%20Imports"
+    )
+
+    return query_url
+
+
+def fetch_data_for_energy_source_and_a_list_of_countries(
+    energy_source: str, countries: List[str]
+) -> pd.DataFrame:
+    """Fetch data from Shift for a specific energy source and a list of countries.
+
+    Parameters
+    ----------
+    energy_source : str
+        Name of energy source (e.g. "coal").
+    countries : list
+        Countries to include in the query.
+
+    Returns
+    -------
+    df : pd.DataFrame
+        Shift data.
+
+    """
+    query_url = prepare_query_url(energy_source=energy_source, countries=countries)
+    soup = BeautifulSoup(requests.get(query_url).content, "html.parser")
+    data = json.loads(
+        soup.find(
+            "script",
+            {"type": "application/json", "id": re.compile(r"^((?!tb-djs).)*$")},
+        ).string
+    )
+
+    fields = data["props"]["apolloState"]
+    elements = {}
+    years = []
+    for key in list(fields):
+        if ("name" in fields[key]) and ("data" in fields[key]):
+            if fields[key]["name"] in countries:
+                elements[fields[key]["name"]] = fields[key]["data"]["json"]
+        if "categories" in fields[key]:
+            years = fields[key]["categories"]["json"]
+
+    assert all([len(elements[country]) == len(years) for country in elements])
+    # Use years as index and elements (data for each country) as columns.
+    df = pd.DataFrame(elements, index=years)
+
+    # Rearrange dataframe for convenience.
+    df = df.reset_index().rename(columns={"index": "year"}).astype({"year": int})
+
+    return df
+
+
+def fetch_all_data_for_energy_source(energy_source: str) -> pd.DataFrame:
+    """Fetch all data for a specific energy source and all countries.
+
+    The list of countries is defined above, in SHIFT_COUNTRIES.
+
+    Parameters
+    ----------
+    energy_source : str
+        Name of energy source (e.g. "coal").
+
+    Returns
+    -------
+    combined : pd.DataFrame
+        Data for a specific energy source and all countries.
+
+    """
+    # Split list of countries in smaller chunks to avoid errors when requesting data.
+    n_chunks = int(len(SHIFT_COUNTRIES) / MAX_NUM_COUNTRIES_PER_QUERY) + 1
+    # Create chunks of country names.
+    countries_chunks = np.array_split(SHIFT_COUNTRIES, n_chunks)
+    dfs = []
+    for countries_chunk in tqdm(countries_chunks, desc="Subset of countries"):
```

Comment:
`tqdm` needs `file=sys.stdout` argument, otherwise it's gonna write into stderr which will trigger slack alerts.